lib/publiccloud/provider.pm

```diff
@@ -43,6 +43,27 @@ sub init {
     $self->create_ssh_key();
 }
 
+=head2 product_openqa_compose
+
+    Call: $self->product_openqa_compose();
+
+    openqa test product name composition, based on basic product parameters values and eventual modifiers.
+
+    Returns a string-name containing distri, version, flavor, arch.
+
+=cut
+
+sub product_openqa_compose {
```

Comment:
```suggestion
sub generate_basename {
```
:thinking:  I am not sure just an idea , but for me it is more verbose than "product_openqa_compose"

Reply:
ok :D